### DISCUSSION TITLE
Is useRouter from next/navigation not supposed to work in async pages?

### DISCUSSION THREAD

**Discussion by omznc** [vercel/next.js#43924] [Help] [answered]
I'm trying to use `useRouter()` in the `app `directory, but it only works if the page isn't `async`.
This works
```ts
import { useRouter } from "next/navigation";

export default function Page() {
    const router = useRouter();
    // Etc...
}
```

But this
```ts
import { useRouter } from "next/navigation";

export default async function Page() {
    const router = useRouter();
    // Etc...
}
```
fails to render with `Error: Cannot read properties of null (reading 'useContext')`

Am I doing something wrong here?

**Comment by icyJoseph** [comment, marked as answer]
`useRouter` from  `next/navigation` is a client component hook: https://beta.nextjs.org/docs/api-reference/use-router
